app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -97,68 +120,138 @@ const MapImpl: React.FunctionComponent = () => {
+                id: featureType,
+              },
+            };
+
+            //dispatch(removeDataset(featuresInState[0].type));
+            dispatch(removeDataset(featureType));
+            dispatch(
+              addDataToMap({
+                datasets,
+                options: {centerMap: true, readOnly: true},
+              })
+            );
           }
         }
 
-        const datasets = {
-          data: Processors.processGeojson({
-            type: "FeatureCollection",
-            features: featuresInState.map(feature => {
-              return {
-                type: "Feature",
-                geometry: wkt.parse(feature.geometry.wkt),
-                properties: feature,
-              };
-            }),
-          }),
-          info: {
-            id: featuresInState[0].type,
-          },
-        };
-        dispatch(
-          addDataToMap({
-            datasets,
-            options: {centerMap: true, readOnly: true},
-          })
-        );
+        break;
+      }
 
+      /*
+      Features in this state have been rendered. In the process of them being added to the map,
+      the variable 'needsFilter' of the appropriate type has been set to true if this was the first time a 
+      feature of that type was added to the map.
+      The step here is to check what filters need to be added and call addFilter if neccessary.
+      */
+      case MapFeatureState.RENDERED: {
+        for (const featureType of Object.values(FeatureType)) {
+          if (state.featuresByType[featureType].needsFilters) {
+            for (var x = 0; x < 3; ++x) {
+              dispatch(
+                addFilter(FeatureType[featureType as keyof typeof FeatureType])
+              );
+            }
+          }
+        }
         break;
       }
 
       case MapFeatureState.CLICKED: {
         /*
-      Features in this state have been clicked on the map. They now need to
-      LOAD in their child features.
+      Features in this state have been clicked on the map. A query for calling features within this feature is called with a limit.
+      This feature may contain even more features than the limit and thus is put into the 'CLICKED AND LOADING' state.
       */
+        totalDataCounter = 0;
+        latestDataCounter = 0;
 
         const clickedUris: string[] = [];
         for (const clickedFeature of featuresInState) {
           if (clickedFeature.type !== FeatureType.Transmission) {
             getFeaturesWithin({
               variables: {
-                query: {withinFeatureUri: clickedFeature.uri},
+                query: {
+                  withinFeatureUri: clickedFeature.uri,
+                },
+                limit: limit,
+                offset: 0,
               },
             });
           }
+
           clickedUris.push(clickedFeature.uri);
         }
+        dispatch(
+          changeMapFeatureState(
+            clickedUris,
+            MapFeatureState.CLICKED_AND_LOADING
```

Comment:
It's not actually CLICKED_AND_LOADING if it's Transmission, though.

Reply:
Also, instead of hard-coding checks against FeatureType in multiple places, please have a function like isExpandable(featureType) or better isExpandable(feature).